networking/p2p/src/main/java/tech/pegasys/teku/networking/p2p/reputation/ReputationManager.java

```diff
@@ -121,5 +141,17 @@ private boolean isLocallyConsideredUnsuitable(
       return locallyInitiated
           && reason.map(r -> !LOCAL_TEMPORARY_DISCONNECT_REASONS.contains(r)).orElse(false);
     }
+
+    public boolean adjustReputation(final ReputationAdjustment effect) {
+      final int newScore =
+          score.updateAndGet(
+              current -> Math.min(MAX_REPUTATION_SCORE, current + effect.getScoreChange()));
+      final boolean shouldDisconnect = newScore <= DISCONNECT_THRESHOLD;
```

Comment:
we create 'shouldDisconnect, set unsuitable to the value of disconnect, return disconnect - maybe just store the value in unsuitable and return that?

Reply:
The peer may have already been marked unsuitable because of repeated failures to connect (e.g if the node is behind a firewall so can connect out but can't accept incoming connections).